src/commands/generate/types.ts

```diff
@@ -80,18 +90,40 @@ export default class GenerateTypes extends Base {
         }
     }
 
-    private getTypesString(variables: Variable[], react: boolean, oldRepos: boolean): string {
+    private async getTypesString(
+        variables: Variable[],
+        orgMembers: OrganizationMember[],
+        react: boolean,
+        oldRepos: boolean
+    ): Promise<string> {
+        const typePromises = variables.map((variable) => this.getTypeDefinitionLine(variable, orgMembers))
+        const typeLines = await Promise.all(typePromises)
         let types = (react ? reactImports(oldRepos) : '') +
             'type DVCJSON = { [key: string]: string | boolean | number }\n\n' +
             'export type DVCVariableTypes = {\n' +
-            `${variables.map((variable) => `    ${this.getVariableKeyAndtype(variable)}`).join('\n')}` +
+            typeLines.join('\n') +
             '\n}'
         if (react) {
             types += reactOverrides
         }
         return types
     }
 
+    private async getTypeDefinitionLine(variable: Variable, orgMembers: OrganizationMember[]): Promise<string> {
+        const { flags } = await this.parse(GenerateTypes)
```

Comment:
Does it make sense to just pass the flags around this these functions rather than having to parse them out 2-3 times?

Reply:
I personally don't like the idea of passing the args around cause it clutters the definitions/calls and makes it more likely that someone will mistakenly pass the args in the wrong positions. Like some of the methods are gonna end up having lots of args i.e. `getTypesString` with 6 which is too much imo

Reply:
was even considering changing `getTypesString` to parse as well